templates/activity_stream/tile_impressions/view.sql

```diff
@@ -0,0 +1,82 @@
+CREATE OR REPLACE VIEW
+  `moz-fx-data-shared-prod.activity_stream.tile_impressions`
+AS
+WITH unioned AS (
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.activity_stream_stable.impression_stats_v1`
+  UNION ALL
+  SELECT
+    *
+  FROM
+    `moz-fx-data-shared-prod.activity_stream_stable.impression_stats_v1`
+  WHERE
+    DATE(submission_timestamp) = CURRENT_DATE()
+),
+impression_data AS (
+  SELECT
+    *
+  FROM
+    unioned
+  WHERE
+    -- make sure data is valid/non-empty
+    ARRAY_LENGTH(tiles) >= 1
+)
+SELECT
+  -- truncate timestamp to seconds
+  TIMESTAMP_TRUNC(submission_timestamp, SECOND) AS submission_timestamp,
```

Comment:
BigQuery needs to be able to push down filters on `submission_timestamp` to the underlying tables to limit the number of day partitions we scan. The base query used `UNIX_SECONDS` here which yields an integer; BigQuery is not able to infer partitions across that type transformation, but it _is_ able to infer partitions through the `TIMESTAMP_TRUNC` function.

Reply:
Amazing, thank you for the change.